File: offer/1.cpp
class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
        map<int,int> cnt;
        for(int i=0; i<nums.size(); ++i) {
            if(cnt.find(nums[i]) == cnt.end()) {
                cnt[nums[i]] = 1;
            }
            else 
                return nums[i];
        }
        return 0;
    }
};

File: offer_js/1.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findRepeatNumber = function(nums) {
    var map = new Map();
    for(num of nums) {
        if(map.has(num))
            return num;
        else
            map.set(num,1);
    }
    return null;
};